__tests__/index.test.js

```diff
@@ -1,80 +1,101 @@
+            expect.objectContaining({
+              description: expect.any(String),
+              slug: expect.any(String),
             })
-        })
-    })
-})
+          );
+        });
+      });
+  });
+});
 
-describe('GET /api', () => {
-    test('GET 200: returns object describing all available endpoints', () => {
-        return request(app)
-        .get("/api")
-        .expect(200)
-        .then(({body}) => {
-            expect(body.endpoints).toEqual(endpoints)
-        })
-    });
-})
+describe("GET /api", () => {
+  test("GET 200: returns object describing all available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.endpoints).toEqual(endpoints);
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("GET 200: returns article with the article ID passed", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article[0]).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        });
+      });
+  });
+  test("GET 404: returns correct error message if passed valid but non-existant article ID number", () => {
+    return request(app)
+      .get("/api/articles/9999")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No Article Found");
+      });
+  });
+  test("GET 400: returns correct error message if passed invalid article ID number", () => {
+    return request(app)
+      .get("/api/articles/bananas")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
 
-describe('GET /api/articles/:article_id', () => {
-    test('GET 200: returns article with the article ID passed', () => {
+describe("GET /api/articles", () => {
+    test("GET 200: returns an array of article objects", () => {
         return request(app)
-        .get("/api/articles/1")
+        .get("/api/articles")
         .expect(200)
-        .then(({body}) => {
-            expect(body.article[0]).toMatchObject(
-                {
-                    article_id: expect.any(Number),
-                    title: expect.any(String),
-                    topic: expect.any(String),
-                    author: expect.any(String),
-                    body: expect.any(String),
-                    created_at: expect.any(String),
-                    votes: expect.any(Number),
-                    article_img_url: expect.any(String),
-                }
-            )
-            console.log(body)
+        .then(({body: {articles}}) => {
+            expect(articles.length = 13)
```

Comment:
This expect statement isnt finished but is very important so I am glad it was included.

Reply:
Do I need to find a way to make it dynamic? So that if new articles are added, it will also include the new article?

Reply:
This expect statement isnt checking anything, by using a single = you are attempting to assign or re-assign something. Here it looks like you are trying to re-assign the length of an array which isnt checking its length. I would recommending looking into the .toBeGreaterThan matcher for this.

Reply:
Oh I see, if I use .toBeGreaterThan would I also need to use .toBeLessThan? So if I was to do .toBeGreaterThan(12) and .toBeLessThan(14) as I am testing if the length is equal to 13?
Alternatively, if I was to use strict equality === would that also work?